connect/runtime/src/main/java/org/apache/kafka/connect/runtime/Worker.java

```diff
@@ -1247,6 +1257,257 @@ void sourceConnectorOffsets(String connName, ConnectorOffsetBackingStore offsetS
         });
     }
 
+    /**
+     * Alter a connector's offsets.
+     *
+     * @param connName the name of the connector whose offsets are to be altered
+     * @param connectorConfig the connector's configurations
+     * @param offsets  a mapping from partitions to offsets that need to be overwritten
```

Comment:
Nits:

1. Might help to be explicit about what "partitions" refer to here
2. We aren't necessarily overwriting here (it's possible that no offset exists for one or more of the specified partitions)

```suggestion
     * @param offsets  a mapping from partitions (either source partitions, or Kafka topic partitions) to offsets that need to be written
```